src/panels/lovelace/cards/hui-entity-button-card.ts

```diff
@@ -29,38 +35,42 @@ interface Config extends LovelaceConfig {
 class HuiEntityButtonCard extends hassLocalizeLitMixin(LitElement)
   implements LovelaceCard {
   public hass?: HomeAssistant;
-  protected config?: Config;
+  private _config?: Config;
 
   static get properties(): PropertyDeclarations {
     return {
       hass: {},
-      config: {},
+      _config: {},
     };
   }
 
-  public getCardSize() {
+  public getCardSize(): number {
     return 2;
   }
 
-  public setConfig(config: Config) {
+  public setConfig(config: Config): void {
     if (!isValidEntityId(config.entity)) {
       throw new Error("Invalid Entity");
     }
 
-    this.config = { theme: "default", ...config };
+    this._config = { theme: "default", ...config };
+  }
 
-    if (this.hass) {
-      this.requestUpdate();
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    if (changedProps.get("hass")) {
+      return (
+        (changedProps.get("hass") as any).states[this._config!.entity] !==
+        this.hass!.states[this._config!.entity]
+      );
     }
+    return (changedProps as unknown) as boolean;
   }
 
-  protected render() {
-    if (!this.config) {
+  protected render(): TemplateResult {
+    if (!this._config) {
       return html``;
     }
-    const stateObj = this.hass!.states[this.config.entity];
-
-    applyThemesOnElement(this, this.hass!.themes, this.config.theme);
+    const stateObj = this.hass!.states[this._config.entity];
```

Comment:
Only ever use `!` if you are 100% sure that this method is always called when this variable is available. Inside `render` you cannot be sure of this.